metric/metric.go

```diff
@@ -9,19 +9,85 @@ import (
 )
 
 var (
+	// Check if the metric is initialized
+	initializedMetrics bool
+
+	// The prefix of the metrics
+	namespace string
+
 	// This will be initialized once PublishMetricsForEndpoint.
 	// The reason why we're doing this is that if metrics are disabled, we don't want to initialize it unnecessarily.
-	resultCount *prometheus.CounterVec = nil
+	resultTotal                              *prometheus.CounterVec
+	resultDurationSeconds                    *prometheus.GaugeVec
+	resultConnectedTotal                     *prometheus.CounterVec
+	resultDNSReturnCodeTotal                 *prometheus.CounterVec
+	resultHTTPStatusCodeTotal                *prometheus.CounterVec
+	resultSSLLastChainExpiryTimestampSeconds *prometheus.GaugeVec
 )
 
+func ensurePrometheusMetrics() {
+	if !initializedMetrics {
+		initializedMetrics = true
```

Comment:
Could you rename `ensurePrometheusMetrics` to `initializePrometheusMetrics` and move the 
`if !initializedMetrics {` statement in the `PublishMetricsForEndpoint` function instead?

e.g.:
```go
func PublishMetricsForEndpoint(endpoint *core.Endpoint, result *core.Result) {
    if !initializedMetrics {
        initializePrometheusMetrics()
        initializedMetrics = true
    }
    // ...
```